plugins/kubernetes/app/models/kubernetes/template_filler.rb

```diff
@@ -240,9 +243,25 @@ def set_secret_puller
     # see http://kubernetes.io/docs/user-guide/production-pods/#handling-initialization
     def modify_init_container
       key = Kubernetes::Api::Pod::INIT_CONTAINER_KEY
-      init_containers = JSON.parse(annotations[key] || '[]', symbolize_names: true)
-      yield init_containers
-      annotations[key] = JSON.pretty_generate(init_containers) if init_containers.any?
+      @init_containers = init_containers
+
+      yield
+
+      if init_containers_in_beta?
+        pod_template.dig_set([:spec, :initContainers], JSON.pretty_generate(@init_containers)) if @init_containers.any?
+        annotations.delete(key)
+      else
+        annotations[key] = JSON.pretty_generate(@init_containers) if @init_containers.any?
+        pod_template[:spec].delete(:initContainers)
+      end
+    end
+
+    def init_containers_in_beta?
+      version = JSON.parse(
+        @doc.deploy_group.kubernetes_cluster.client.create_rest_client('version').get.body
```

Comment:
Let's cache this value for each kubernetes cluster, at least for a minute or two.

Otherwise we'll end up calling out to the cluster each time we render a template that uses secrets.

Reply:
Also extract to a method, so you could call `@doc.deploy_group.kubernetes_cluster.server_version`, which would have the caching logic, and return a `Gem::Version` object.